var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -58,3 +62,36 @@ def cmake_args(self):
         ]
 
         return args
+
+    @run_after("install")
+    def setup_build_tests(self):
+        launcher = self.mpi_launcher()
+
+        exe = os.path.basename(launcher.path)
+        filename = "samples/sync-write-1d-contig-contig.json"
+        filter_file(f"{exe}", f"{launcher}", filename)
```

Comment:
Does the command/executable in `samples/sync-write-1d-contig-contig.json` ever change?  Or is it fixed at, say `mpirun`? 

Regardless, you'll want to use whatever is already in the file as the first argument.  Otherwise, this line will only work if the file's executable/command actually matches the name of the launcher.

Reply:
This last push should cover this.